src/plots/errorBarPlot.ts

```diff
@@ -0,0 +1,115 @@
+///<reference path="../reference.ts" />
+
+module Plottable {
+export module Plots {
+  export class ErrorBar<X, Y> extends XYPlot<X, Y> {
+
+    private _tickLength = 10;
+
+    constructor() {
+      super();
+      this.addClass("error-bar-plot");
+    }
+
+    /**
+     * Sets the tick length
+     */
+    public tickLength(): number;
+    public tickLength(tickLength: number): ErrorBar<X, Y>;
+    public tickLength(tickLength?: number): any {
+      if (tickLength == null) {
+        return this._tickLength;
+      }
+      this._tickLength = tickLength;
+    }
+
+    /**
+     * Gets the AccessorScaleBinding for X2.
+     */
+    public x2(): AccessorScaleBinding<X, number>;
+    /**
+     * Sets X2 to a constant number or the result of an Accessor.
+     * If a Scale has been set for X, it will also be used to scale X2.
+     *
+     * @param {number|Accessor<number>|X|Accessor<X>} x2
+     * @returns {Plots.ErrorBar} The calling Error Bar Plot.
+     */
+    public x2(x2: number | Accessor<number> | X | Accessor<X>): ErrorBar<X, Y>;
+    public x2(x2?: number | Accessor<number> | X | Accessor<X>): any {
+      if (x2 == null) {
+        return this._propertyBindings.get("x2");
+      }
+      var xBinding = this.x();
+      var xScale = xBinding && xBinding.scale;
+      this._bindProperty("x2", x2, xScale);
+      this.render();
+      return this;
+    }
+
+    /**
+     * Gets the AccessorScaleBinding for Y2.
+     */
+    public y2(): AccessorScaleBinding<Y, number>;
+    /**
+     * Sets Y2 to a constant number or the result of an Accessor.
+     * If a Scale has been set for Y, it will also be used to scale Y2.
+     *
+     * @param {number|Accessor<number>|Y|Accessor<Y>} y2
+     * @returns {Plots.ErrorBar} The calling Error Bar Plot.
+     */
+    public y2(y2: number | Accessor<number> | Y | Accessor<Y>): ErrorBar<X, Y>;
+    public y2(y2?: number | Accessor<number> | Y | Accessor<Y>): any {
+      if (y2 == null) {
+        return this._propertyBindings.get("y2");
+      }
+      var yBinding = this.y();
+      var yScale = yBinding && yBinding.scale;
+      this._bindProperty("y2", y2, yScale);
+      this.render();
+      return this;
+    }
+
+    protected _additionalPaint(time: number) {
+      super._additionalPaint(time);
+      console.log(this._tickLength);
+      if (this._tickLength != null) {
+        var lowerBars = this._renderArea.append("g").classed("error-lower-bars", true);
+        var upperBars = this._renderArea.append("g").classed("error-upper-bars", true);
```

Comment:
The logic here doesn't seem quite right. For example, repeated calls to `_additionalPaint` will generate more `<g>`s every time.

In addition, we now have the problem that the `Entity` for a given point contains only the center part of the error bar and not the two ends.